app/queries/intersection_query.rb

```diff
@@ -0,0 +1,25 @@
+class IntersectionQuery
```

Comment:
This appears to be a [lazy class](https://sourcemaking.com/refactoring/smells/lazy-class) and can probably be combined with SurveyResponse.